### PR TITLE
WIP pl_proxy

### PR DESCRIPTION
**Pull request by snopoke**
@czue @dannyroberts @benrudolph @NoahCarnahan 

first commit is just converting to multi-db setup

### CODE REVIEW

File: prototype/instance/config.py [hunk 1]
```diff
@@ -6,7 +6,12 @@
 SQLALCHEMY_ECHO = False
 
 SECRET_KEY = 'testkey'
-SQLALCHEMY_DATABASE_URI = 'postgresql://postgres:@localhost/receiver'
+
+SQLALCHEMY_DATABASE_URI= 'postgresql://skelly@localhost/receiver_proxy'
```
Comment (czue) [inline prototype/instance/config.py]: was this intentional?

**Comment by czue** [comment]
this is sweet!

File: prototype/instance/config.py (hunk 1 above)
Comment (snopoke) [inline prototype/instance/config.py]: yup, main DB points to proxy so all queries get routed through it

File: prototype/migrations/script.py.mako [hunk 2]
```diff
@@ -9,14 +12,35 @@ Create Date: ${create_date}
...
+
+<%
+    from flask import current_app
+    db_names = [''] + list(current_app.config.get("SQLALCHEMY_BINDS").keys())
+%>
+
+## generate an "upgrade_<xyz>() / downgrade_<xyz>()" function
+## for each database name in the ini file.
+
+% for db_name in db_names:
```
Comment (dannyroberts) [inline prototype/migrations/script.py.mako:37]: Isn't this going to be a problem, having migration file contain separate migrations for each database? I.e. if `SQLALCHEMY_BINDS` ever changes, or even differs between production setups or even local setups, then this approach won't work, right?

Comment (snopoke) [inline prototype/migrations/script.py.mako:37]: yea, but I wasn't worrying about that here

File: prototype/instance/config.py (hunk 1 above)
Comment (czue) [inline prototype/instance/config.py]: (i meant the part where you changed `postgres` to `skelly`)

**Comment by snopoke** [comment]
This is as far as I'm going to take this. Form submissions and restore working.

Not all DB access is going the the proxy, only forms, cases and case indices.

**Pull request closed without merging**